### ISSUE TITLE
Conda Installation Solving Environment Failed

### ISSUE THREAD

**Issue by M-Harrington** [conda-forge/rasterio-feedstock#166]
Issue: Identical to #159, but with included diagnostic information. Running the full conflict investigation returns a humongous amount of text, so I can share it by request but I won't post here by default.

``` 
(base) C:\Users\Matttt>conda config --add channels conda-forge
Warning: 'conda-forge' already in 'channels' list, moving to the top

(base) C:\Users\Matttt>conda install rasterio
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: 
```

<br/>
Environment (<code>conda list</code>):
<details>
```
(base) C:\Users\Matttt>conda config --show-sources
==> C:\Users\Matttt\.condarc <==
ssl_verify: True
channels:
  - conda-forge
  - defaults
```

```
$ conda list
(base) C:\Users\Matttt>conda list --show-channel-urls
# packages in environment at C:\Users\Matttt\anaconda3:
#
# Name                    Version                   Build  Channel
_ipyw_jlab_nb_ext_conf    0.1.0                    py37_0    defaults
alabaster                 0.7.12                   py37_0    defaults
anaconda                  2020.02                  py37_0    defaults
anaconda-client           1.7.2                    py37_0    defaults
anaconda-navigator        1.9.12                   py37_0    defaults
anaconda-project          0.8.4                      py_0    defaults
argh                      0.26.2                   py37_0    defaults
asn1crypto                1.3.0                    py37_0    conda-forge
astroid                   2.3.3                    py37_0    defaults
astropy                   4.0              py37he774522_0    defaults
atomicwrites              1.3.0                    py37_1    defaults
attrs                     19.3.0                     py_0    conda-forge
autopep8                  1.4.4                      py_0    defaults
babel                     2.8.0                      py_0    conda-forge
backcall                  0.1.0                    py37_0    defaults
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
backports.shutil_get_terminal_size 1.0.0                    py37_2    defaults
backports.tempfile        1.0                        py_0    conda-forge
backports.weakref         1.0.post1       py37hc8dfbb8_1001    conda-forge
bcrypt                    3.1.7            py37he774522_0    defaults
beautifulsoup4            4.8.2                    py37_0    defaults
bitarray                  1.2.1            py37he774522_0    defaults
bkcharts                  0.2                      py37_0    defaults
blas                      1.0                         mkl    defaults
bleach                    3.1.0                    py37_0    defaults
blosc                     1.16.3               h7bd577a_0    defaults
bokeh                     1.4.0                    py37_0    conda-forge
boto                      2.49.0                   py37_0    defaults
bottleneck                1.3.2            py37h2a96729_0    defaults
bzip2                     1.0.8                he774522_0    defaults
ca-certificates           2020.1.1                      0    defaults
certifi                   2019.11.28               py37_0    defaults
cffi                      1.14.0           py37h7a1dbc1_0    defaults
cftime                    1.1.3            py37h44b1f71_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
click                     7.0                      py37_0    defaults
cloudpickle               1.3.0                      py_0    defaults
clyent                    1.2.2                    py37_1    defaults
colorama                  0.4.3                      py_0    conda-forge
comtypes                  1.1.7                    py37_0    defaults
conda                     4.8.3            py37hc8dfbb8_1    conda-forge
conda-build               3.18.11                  py37_0    defaults
conda-env                 2.6.0                         1    defaults
conda-package-handling    1.6.0            py37h702c6c1_2    conda-forge
conda-verify              3.4.2                      py_1    defaults
console_shortcut          0.1.1                         4    defaults
contextlib2               0.6.0.post1                py_0    conda-forge
cryptography              2.8              py37h7a1dbc1_0    defaults
curl                      7.68.0               h2a8f88b_0    defaults
cycler                    0.10.0                   py37_0    defaults
cython                    0.29.15          py37ha925a31_0    defaults
cytoolz                   0.10.1           py37he774522_0    defaults
dask                      2.11.0                     py_0    defaults
dask-core                 2.11.0                     py_0    defaults
decorator                 4.4.1                      py_0    defaults
defusedxml                0.6.0                      py_0    conda-forge
diff-match-patch          20181111                   py_0    conda-forge
distributed               2.11.0                   py37_0    defaults
docutils                  0.16                     py37_0    conda-forge
eli5                      0.10.1                   py37_1    conda-forge
entrypoints               0.3                      py37_0    defaults
et_xmlfile                1.0.1                    py37_0    defaults
fastcache                 1.1.0            py37he774522_0    defaults
filelock                  3.0.12                     py_0    defaults
flake8                    3.7.9                    py37_0    conda-forge
flask                     1.1.1                      py_0    defaults
freetype                  2.9.1                ha9979f8_1    defaults
fsspec                    0.6.2                      py_0    defaults
future                    0.18.2                   py37_0    conda-forge
get_terminal_size         1.0.0                h38e98db_0    defaults
gevent                    1.4.0            py37he774522_0    defaults
glob2                     0.7                        py_0    conda-forge
graphviz                  2.38.0            h6538335_1011    conda-forge
greenlet                  0.4.15           py37hfa6e2cd_0    conda-forge
h5py                      2.10.0           py37h5e291fa_0    defaults
hdf4                      4.2.13               h712560f_2    defaults
hdf5                      1.10.4               h7ebc959_0    defaults
heapdict                  1.0.1                      py_0    conda-forge
html5lib                  1.0.1                    py37_0    defaults
hypothesis                5.5.4                      py_0    defaults
icc_rt                    2019.0.0             h0cc432a_1    defaults
icu                       58.2                 ha66f8fd_1    defaults
idna                      2.8                      py37_0    defaults
imageio                   2.6.1                    py37_0    conda-forge
imagesize                 1.2.0                      py_0    conda-forge
importlib_metadata        1.5.0                    py37_0    conda-forge
intel-openmp              2020.0                      166    defaults
intervaltree              3.0.2                      py_0    conda-forge
ipykernel                 5.1.4            py37h39e3cac_0    defaults
ipython                   7.12.0           py37h5ca1d4c_0    defaults
ipython_genutils          0.2.0                    py37_0    defaults
ipywidgets                7.5.1                      py_0    conda-forge
isort                     4.3.21                   py37_0    conda-forge
itsdangerous              1.1.0                    py37_0    defaults
jdcal                     1.4.1                      py_0    conda-forge
jedi                      0.14.1                   py37_0    defaults
jinja2                    2.11.1                     py_0    defaults
joblib                    0.14.1                     py_0    defaults
jpeg                      9b                   hb83a4c4_2    defaults
json5                     0.9.1                      py_0    defaults
jsonschema                3.2.0                    py37_0    conda-forge
jupyter                   1.0.0                    py37_7    defaults
jupyter_client            5.3.4                    py37_0    conda-forge
jupyter_console           6.1.0                      py_0    defaults
jupyter_core              4.6.1                    py37_0    defaults
jupyterlab                1.2.6              pyhf63ae98_0    defaults
jupyterlab_server         1.0.6                      py_0    defaults
keyboard                  0.13.5                   pypi_0    pypi
keyring                   21.1.0                   py37_0    defaults
kiwisolver                1.1.0            py37ha925a31_0    defaults
krb5                      1.17.1               hc04afaa_0    defaults
lazy-object-proxy         1.4.3            py37he774522_0    defaults
libarchive                3.3.3                h0643e63_5    defaults
libcurl                   7.68.0               h2a8f88b_0    defaults
libiconv                  1.15                 h1df5818_7    defaults
liblief                   0.9.0                ha925a31_2    defaults
libnetcdf                 4.7.3                h1302dcc_0    defaults
libpng                    1.6.37               h2a8f88b_0    defaults
libsodium                 1.0.16               h9d3ae62_0    defaults
libspatialindex           1.9.3                h33f27b4_0    defaults
libssh2                   1.8.2                h7a1dbc1_0    defaults
libtiff                   4.1.0                h56a325e_0    defaults
libxml2                   2.9.9                h464c3ec_0    defaults
libxslt                   1.1.33               h579f668_0    defaults
llvmlite                  0.31.0           py37ha925a31_0    defaults
locket                    0.2.0                    py37_1    defaults
lxml                      4.5.0            py37h1350720_0    defaults
lz4-c                     1.8.1.2              h2fa13f4_0    defaults
lzo                       2.10                 h6df0209_2    defaults
m2w64-gcc-libgfortran     5.3.0                         6    defaults
m2w64-gcc-libs            5.3.0                         7    defaults
m2w64-gcc-libs-core       5.3.0                         7    defaults
m2w64-gmp                 6.1.0                         2    defaults
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    defaults
markupsafe                1.1.1            py37he774522_0    defaults
matplotlib                3.1.3                    py37_0    defaults
matplotlib-base           3.1.3            py37h64f37c6_0    defaults
mccabe                    0.6.1                    py37_1    defaults
menuinst                  1.4.16           py37he774522_0    defaults
mistune                   0.8.4            py37he774522_0    defaults
mkl                       2020.0                      166    defaults
mkl-service               2.3.0            py37hb782905_0    defaults
mkl_fft                   1.0.15           py37h14836fe_0    defaults
mkl_random                1.1.0            py37h675688f_0    defaults
mock                      4.0.1                      py_0    defaults
more-itertools            8.2.0                      py_0    defaults
mpmath                    1.1.0                    py37_0    defaults
msgpack-python            0.6.1            py37h74a9793_1    defaults
msys2-conda-epoch         20160418                      1    defaults
multipledispatch          0.6.0                    py37_0    defaults
navigator-updater         0.2.1                    py37_0    defaults
nbconvert                 5.6.1                    py37_0    conda-forge
nbformat                  5.0.4                      py_0    defaults
netcdf4                   1.5.3            py37h012c1a0_0    defaults
networkx                  2.4                        py_0    conda-forge
nltk                      3.4.5                    py37_0    defaults
nose                      1.3.7                    py37_2    conda-forge
notebook                  6.0.3                    py37_0    conda-forge
numba                     0.48.0           py37h47e9c7a_0    defaults
numexpr                   2.7.1            py37h25d0782_0    defaults
numpy                     1.18.1           py37h93ca92e_0    defaults
numpy-base                1.18.1           py37hc3f5095_1    defaults
numpydoc                  0.9.2                      py_0    defaults
olefile                   0.46                     py37_0    defaults
openpyxl                  3.0.3                      py_0    conda-forge
openssl                   1.1.1d               he774522_4    defaults
packaging                 20.1                       py_0    defaults
pandas                    1.0.1            py37h47e9c7a_0    defaults
pandoc                    2.2.3.2                       0    defaults
pandocfilters             1.4.2                    py37_1    defaults
paramiko                  2.7.1                      py_0    defaults
parso                     0.5.2                      py_0    defaults
partd                     1.1.0                      py_0    conda-forge
path                      13.1.0                   py37_0    defaults
path.py                   12.4.0                        0    conda-forge
pathlib2                  2.3.5                    py37_0    conda-forge
pathtools                 0.1.2                      py_1    conda-forge
patsy                     0.5.1                    py37_0    defaults
pep8                      1.7.1                    py37_0    defaults
pexpect                   4.8.0                    py37_0    conda-forge
pickleshare               0.7.5                    py37_0    defaults
pillow                    7.0.0            py37hcc1f983_0    defaults
pip                       20.0.2                   py37_1    conda-forge
pkginfo                   1.5.0.1                  py37_0    defaults
pluggy                    0.13.1                   py37_0    defaults
ply                       3.11                     py37_0    defaults
powershell_shortcut       0.0.1                         3    defaults
prometheus_client         0.7.1                      py_0    defaults
prompt_toolkit            3.0.3                      py_0    defaults
psutil                    5.6.7            py37he774522_0    defaults
py                        1.8.1                      py_0    conda-forge
py-lief                   0.9.0            py37ha925a31_2    defaults
pyautogui                 0.9.48                   py37_0    conda-forge
pycodestyle               2.5.0                    py37_0    defaults
pycosat                   0.6.3            py37he774522_0    defaults
pycparser                 2.19                     py37_0    defaults
pycrypto                  2.6.1            py37hfa6e2cd_9    defaults
pycurl                    7.43.0.5         py37h7a1dbc1_0    defaults
pydocstyle                4.0.1                      py_0    defaults
pyflakes                  2.1.1                    py37_0    defaults
pygments                  2.5.2                      py_0    defaults
pylint                    2.4.4                    py37_0    defaults
pymsgbox                  1.0.7                      py_0    conda-forge
pynacl                    1.3.0            py37h62dcd97_0    defaults
pyodbc                    4.0.30           py37ha925a31_0    defaults
pyopenssl                 19.1.0                   py37_0    conda-forge
pyparsing                 2.4.6                      py_0    defaults
pyqt                      5.9.2            py37h6538335_2    defaults
pyreadline                2.1                      py37_1    defaults
pyrsistent                0.15.7           py37he774522_0    defaults
pyscreeze                 0.1.26                     py_0    conda-forge
pysocks                   1.7.1                    py37_0    conda-forge
pytables                  3.6.1            py37h1da0976_0    defaults
pytest                    5.3.5                    py37_0    defaults
pytest-arraydiff          0.3              py37h39e3cac_0    defaults
pytest-astropy            0.8.0                      py_0    defaults
pytest-astropy-header     0.1.2                      py_0    conda-forge
pytest-doctestplus        0.5.0                      py_0    defaults
pytest-openfiles          0.4.0                      py_0    conda-forge
pytest-remotedata         0.3.2                    py37_0    defaults
python                    3.7.6                h60c2a47_2    defaults
python-dateutil           2.8.1                      py_0    conda-forge
python-graphviz           0.14               pyh9f0ad1d_0    conda-forge
python-jsonrpc-server     0.3.4                      py_0    conda-forge
python-language-server    0.31.7                   py37_0    defaults
python-libarchive-c       2.8                     py37_13    defaults
python_abi                3.7                     1_cp37m    conda-forge
pytweening                1.0.3                      py_0    conda-forge
pytz                      2019.3                     py_0    conda-forge
pywavelets                1.1.1            py37he774522_0    defaults
pywin32                   227              py37he774522_1    defaults
pywin32-ctypes            0.2.0                 py37_1000    conda-forge
pywinpty                  0.5.7                    py37_0    conda-forge
pyyaml                    5.3              py37he774522_0    defaults
pyzmq                     18.1.1           py37ha925a31_0    defaults
qdarkstyle                2.8                        py_0    defaults
qt                        5.9.7            vc14h73c81de_0    defaults
qtawesome                 0.6.1                      py_0    defaults
qtconsole                 4.6.0                      py_1    defaults
qtpy                      1.9.0                      py_0    conda-forge
requests                  2.22.0                   py37_1    conda-forge
rope                      0.16.0                     py_0    defaults
rtree                     0.9.3            py37h21ff451_0    defaults
ruamel_yaml               0.15.87          py37he774522_0    defaults
scikit-image              0.16.2           py37h47e9c7a_0    defaults
scikit-learn              0.22.1           py37h6288b17_0    defaults
scipy                     1.4.1            py37h9439919_0    defaults
seaborn                   0.10.0                     py_0    defaults
send2trash                1.5.0                    py37_0    defaults
setuptools                45.2.0                   py37_0    defaults
simplegeneric             0.8.1                    py37_2    defaults
singledispatch            3.4.0.3                  py37_0    defaults
sip                       4.19.8           py37h6538335_0    defaults
six                       1.14.0                   py37_0    conda-forge
snappy                    1.1.7                h777316e_3    defaults
snowballstemmer           2.0.0                      py_0    conda-forge
sortedcollections         1.1.2                    py37_0    defaults
sortedcontainers          2.1.0                    py37_0    defaults
soupsieve                 1.9.5                    py37_0    defaults
sphinx                    2.4.0                      py_0    defaults
sphinxcontrib             1.0                      py37_1    defaults
sphinxcontrib-applehelp   1.0.1                      py_0    defaults
sphinxcontrib-devhelp     1.0.1                      py_0    defaults
sphinxcontrib-htmlhelp    1.0.2                      py_0    defaults
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.2                      py_0    defaults
sphinxcontrib-serializinghtml 1.1.3                      py_0    defaults
sphinxcontrib-websupport  1.2.0                      py_0    defaults
spyder                    4.0.1                    py37_0    defaults
spyder-kernels            1.8.1                    py37_0    defaults
sqlalchemy                1.3.13           py37he774522_0    defaults
sqlite                    3.31.1               he774522_0    defaults
statsmodels               0.11.0           py37he774522_0    defaults
sympy                     1.5.1                    py37_0    conda-forge
tabulate                  0.8.7              pyh9f0ad1d_0    conda-forge
tbb                       2020.0               h74a9793_0    defaults
tblib                     1.6.0                      py_0    conda-forge
terminado                 0.8.3                    py37_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
tk                        8.6.8                hfa6e2cd_0    defaults
toolz                     0.10.0                     py_0    conda-forge
tornado                   6.0.3            py37he774522_3    defaults
tqdm                      4.42.1                     py_0    defaults
traitlets                 4.3.3                    py37_0    conda-forge
ujson                     1.35             py37hfa6e2cd_0    defaults
unicodecsv                0.14.1                   py37_0    defaults
urllib3                   1.25.8                   py37_0    defaults
vc                        14.1                 h0510ff6_4    defaults
vs2015_runtime            14.16.27012          hf0eaf9b_1    defaults
watchdog                  0.10.2                   py37_0    conda-forge
wcwidth                   0.1.8                      py_0    defaults
webencodings              0.5.1                    py37_1    defaults
werkzeug                  1.0.0                      py_0    defaults
wheel                     0.34.2                   py37_0    conda-forge
widgetsnbextension        3.5.1                    py37_0    conda-forge
win_inet_pton             1.1.0                    py37_0    conda-forge
win_unicode_console       0.5                      py37_0    conda-forge
wincertstore              0.2                      py37_0    defaults
winpty                    0.4.3                         4    conda-forge
wrapt                     1.11.2           py37he774522_0    defaults
xarray                    0.15.1                     py_0    conda-forge
xlrd                      1.2.0                    py37_0    defaults
xlsxwriter                1.2.7                      py_0    defaults
xlwings                   0.17.1                   py37_0    defaults
xlwt                      1.3.0                    py37_0    defaults
xmltodict                 0.12.0                     py_0    conda-forge
xz                        5.2.4                h2fa13f4_4    defaults
yaml                      0.1.7                hc54c509_2    defaults
yapf                      0.28.0                     py_0    defaults
zeromq                    4.3.1                h33f27b4_3    defaults
zict                      1.0.0                      py_0    defaults
zipp                      2.2.0                      py_0    defaults
zlib                      1.2.11               h62dcd97_3    defaults
zstd                      1.3.7                h508b16e_0    defaults
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
(base) C:\Users\Matttt>conda info

     active environment : base
    active env location : C:\Users\Matttt\anaconda3
            shell level : 1
       user config file : C:\Users\Matttt\.condarc
 populated config files : C:\Users\Matttt\.condarc
          conda version : 4.8.3
    conda-build version : 3.18.11
         python version : 3.7.6.final.0
       virtual packages :
       base environment : C:\Users\Matttt\anaconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\Matttt\anaconda3\pkgs
                          C:\Users\Matttt\.conda\pkgs
                          C:\Users\Matttt\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\Matttt\anaconda3\envs
                          C:\Users\Matttt\.conda\envs
                          C:\Users\Matttt\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.8.3 requests/2.22.0 CPython/3.7.6 Windows/10 Windows/10.0.17763
          administrator : False
             netrc file : None
           offline mode : False
```
</details>

I have also tried updating conda just before and after, and in both situations the same error results.  I have not yet tried to allow-downgrades with conda because I'm not sure what version to downgrade to and I'm worried about affecting other packages. 

Please let me know if there's any other information I could post that would be helpful.

**Comment by snowman2** [comment]
Looks like a lot of `defaults` stuff in there. I would recommend changing `.condarc` to:
```
channels:
  - conda-forge
  - defaults
channel_priority: strict
```

**Comment by M-Harrington** [comment]
Doesn't look like there was an improvement unfortunately

**Comment by ocefpaf** [comment]
> Doesn't look like there was an improvement unfortunately

Please post the new list of package. If they still have `defaults` in them check if you have multiple .condarc files, maybe you edited the wrong one. BTW I even recommend to remove defaults entirely.

**Comment by M-Harrington** [comment]
Defaults were still there for some reason, even though I had double checked that the `.condarc` file was unique.  Unfortunately I couldn't find an easy way to remove the defaults (`conda remove * -c defaults` didn't seem to do the trick), so I uninstalled Anaconda completely and downloaded a fresh install of miniconda.  For the moment everything is working fine, so even though I'm not sure what the conflict was, I've managed to fix the issue.
